src/collectd.conf.pod

```diff
@@ -11600,31 +11600,55 @@ want to set B<metadata.broker.list> to your Kafka broker list.
 =head2 Plugin C<write_redis>
 
 The I<write_redis plugin> submits values to I<Redis>, a data structure server.
+It uses the I<Time series> type provided by the I<RedisTimeSeries> module.
 
 Synopsis:
 
   <Plugin "write_redis">
     <Node "example">
         Host "localhost"
-        Port "6379"
-        Timeout 1000
-        Prefix "collectd/"
-        Database 1
-        MaxSetSize -1
-        MaxSetDuration -1
+        Port 6379
+        Database 0
+        Timeout 1.0
+        Retention 0
         StoreRates true
+        Prefix ""
     </Node>
   </Plugin>
 
-Values are submitted to I<Sorted Sets>, using the metric name as the key, and
-the timestamp as the score. Retrieving a date range can then be done using the
-C<ZRANGEBYSCORE> I<Redis> command. Additionally, all the identifiers of these
-I<Sorted Sets> are kept in a I<Set> called C<collectd/values> (or
-C<${prefix}/values> if the B<Prefix> option was specified) and can be retrieved
-using the C<SMEMBERS> I<Redis> command. You can specify the database to use
-with the B<Database> parameter (default is C<0>). See
-L<http://redis.io/commands#sorted_set> and L<http://redis.io/commands#set> for
-details.
+Metric values are written to I<Time series> keys, using the JSON encoded metric
+identity prefixed by C<metric/> as the key. When a metric is encountered for
+the first time, collectd tries to create a new time series using C<TS.CREATE>.
+This fails if the time series already exists, therefore C<TS.CREATE> errors are
+logged but ignored.
+
+The plugin writes one I<Set> per I<resource> (e.g. host). The set's key is the
+JSON encoded resource attributes, prefixed by C<resource/>. The values are the
+metric IDs as outlined above. The write is only performed when the metric is
+"new", i.e. it is observed for the first time.
+
+Finally, the keys of all resource I<Sets> are written to a global I<Set>,
+called C<resources>. A list of resources can be retrieved using the C<SMEMBERS>
+I<Redis> command. The write is only performed if all metrics in the metric
+family are "new", see above.
+
+Putting it all together: for a metric family with a single metric, the
+following Redis operations are performed when the metric is encountered for the
+first time:
+
+  TS.CREATE metric/<metric_id> RETENTION <Retention> LABELS <name0> <value0> ...
+  TS.ADD metric/<metric_id> <timestamp> <value>
+  SADD resource/<resource_id> metric/<metric_id>
+  SADD resources resource/<resource_id>
+
+On subsequent writes, only the C<TS.ADD> command is performed.
+
+More information about the Redis data types used here is available at
+L<https://redis.io/docs/data-types/timeseries/> and
+L<http://redis.io/commands#set>.
+
+You can specify the database to use with the B<Database> parameter (default is
+C<0>).
```

Comment:
Redis databases are not named?  If that is some kind of DB index, maybe the option name should indicate that too?

Reply:
Yes, apparently. The feature is not documented very well. Almost all I know about it is based on [the `SELECT` command](https://redis.io/commands/select/). (As you can tell, I'm not a Redis expert myself.)

Redis calls this a "database number", so if "Database" it too unspecific, that would be the next best naming. I think that "Database" is fine though.

Reply:
This says it's "Database Index": https://redis.io/commands/select/

Maybe "DBIndex"?

(Do as you wish, I've already approved this, and do not use Redis, so I cannot really say what Redis users expect as terminology.)

Reply:
The text mentions "database 0" and "database zero" multiple times. The index is mentioned in the syntax and to clarify what "database" means in Redis' context. I think Redis users would say "database 1", and the existing config key would be quite natural.